controller/controller.go

```diff
@@ -223,7 +229,7 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 
 // ProcessEvents takes the event channels created by the operatorkit informer
 // and executes the controller's event functions accordingly.
-func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Event, updateChan chan watch.Event, errChan chan error) {
+func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Event, updateChan chan watch.Event, errChan chan error) error {
```

Comment:
This is some other boyscouting so I made this returning an error. In the future this should maybe made private as well since the user does not benefit from it in any way and it is actually quite internal. The improvement here now leads to a better error flow. The `os.Exit(1)` magic is already done in the boot and with this we redirect it to there.